osrdyne/src/config.rs

```diff
@@ -32,6 +34,8 @@ pub struct OsrdyneConfig {
     pub max_length_bytes: Option<usize>,
     pub api_address: String,
     pub extra_lifetime: Option<Duration>,
+    pub service_name: String,
+    pub opentelemetry_endpoint: Option<Url>,
```

Comment:
Do we want/need the `service_name` if the opentelemetry endpoint is not defined 🤔 If not, then maybe this can be done like this?
```suggestion
    pub opentelemetry_config: Option<OpenTelemetryConfig>,
```

with
```rust
struct OpentelemetryConfig {
  service_name: String,
  endpoint: Url
}
```